Description: include/buffer: include <memory>
 to address following FTBFS (under GCC 12):
 .
 /usr/bin/ccache /usr/bin/clang++-13 -DBOOST_ALL_NO_LIB -DBOOST_ASIO_DISABLE_CONCEPTS -DBOOST_ASIO_DISABLE_THREAD_KEYWORD_EXTENSION -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_T$
 In file included from /var/ssd/ceph/src/crimson/os/seastore/seastore_types.cc:4:
 In file included from /var/ssd/ceph/src/crimson/os/seastore/seastore_types.h:14:
 In file included from /var/ssd/ceph/src/include/denc.h:47:
 /var/ssd/ceph/src/include/buffer.h:98:37: error: no template named 'unique_ptr' in namespace 'std'; did you mean 'boost::movelib::unique_ptr'?
 struct unique_leakable_ptr : public std::unique_ptr<T, ceph::nop_delete<T>> {
                                     ^~~~~~~~~~~~~~~
                                     boost::movelib::unique_ptr
 /opt/ceph/include/boost/move/unique_ptr.hpp:354:7: note: 'boost::movelib::unique_ptr' declared here
 class unique_ptr
       ^
Signed-off-by: Kefu Chai <tchaikov@gmail.com>
Author: Kefu Chai <tchaikov@gmail.com>
Origin: upstream, https://github.com/ceph/ceph/commit/7c381ba985bd1398ef7d145cc00fae9d0db510e3
Last-Update: 2022-07-27

--- ceph-16.2.10+ds.orig/src/include/buffer.h
+++ ceph-16.2.10+ds/src/include/buffer.h
@@ -41,6 +41,7 @@
 #include <iosfwd>
 #include <iomanip>
 #include <list>
+#include <memory>
 #include <vector>
 #include <string>
 #if __cplusplus >= 201703L
